src/server/main.py

```diff
@@ -1,46 +1,66 @@
+"""Module enables socket creation and listen to client"""
+import socket
 from utils.serialise import Serialiser
 from utils.decrept import Decryption
 from utils.file_handler import FileHandler
-import socket
+
+
+def convert_file(input_data, format_input):
+    """
+    Helper function that creates a txt file then convertes it to the desired format
+    """
+    txt = FileHandler.create_txt(input_str=input_data)
+    output_file = FileHandler.convert_file(file=txt, format_type=format_input)
+    return output_file
 
 
 def receive_from_client(client_data):
```

Comment:
This function seems to be quite lengthy, maybe it can be broken down into smaller functions? 

- A function to handle decryption
- A function to handle output (printing or file conversion).

Doing this would also make the code more readable and we can avoid nested statements.
It might be a good idea to initialize variables such as "requested_output" at the beginning of the function. Current logic won't create any issues but if the code is later modified we could run into issues since this variable is not created at the beginning of the function.

Seems like there is some code repetition. E.g. when handling "PRINT_OUTPUT" and "TEXT_FILE_OUTPUT". It might be a good idea to refactor this. 
Suggestion: Have perhaps a few helper functions to get rid of that. 
1- def process_client_data(client_data: Dict[str, Any)] -> Union[str, None]:
"""
    Process data sent from the client by deserializing it, optionally decrypting it,
    and then handling the output based on client preferences.

    :param client_data: Serialized data sent from the client.
    :return: Processed output, either a printed message or a file reference.
    """
2- def get_message(deserialised: Dict[str, Any]) -> str:
"""
    Extract the message from the deserialized client data. If the data is encrypted,
    decrypt it before returning.

    :param deserialised: Deserialized client data.
    :return: Decrypted or original message from the client.
    """
3- def handle_output(deserialised: Dict[str, Any)], message: str) -> Union[str, None]:
"""
    Handle the output of the processed client message based on client preferences.
    The message can either be printed or saved to a file in a specified format.

    :param deserialised: Deserialized client data containing output preferences.
    :param message: Processed message to be outputted.
    :return: Printed message or file reference based on client request.
    """

Reply:
Thanks Taylan, this was very helpful.